### ISSUE TITLE
DocCommentLongArraySyntaxSniff loops forever if @endcode is missing in PHP 8.1

### ISSUE THREAD

**Issue by claudiu-cristea** [squizlabs/PHP_CodeSniffer#3654]
**Describe the bug**

Happens after updating to PHP 8.1. In docblock a sample code is opened with `@code` but it's not closed, as expected, with `@endcode`. Let's say it was closed with `@code`. In this case, this line will return `false`:
```
$codeEnd = $phpcsFile->findNext(T_DOC_COMMENT_TAG, ($codeStart + 1), $commentEnd, false, '@endcode');
```
and the `do {} while ()` block will loop forever.

**Code sample**
```php
<?php
  /**
   * The structure of the array is:
   * @code
   * $workflow_array = [
   *   'entity_state' => [
   *     'user' => [
   *       'transition1',
   *       'transition2',
   *     ],
   *   ],
   * ];
   * @code
   */
```

**Custom ruleset**

N/A

**To reproduce**

See th code sample.

PHPCS output: The process gets stuck in DocCommentLongArraySyntaxSniff

**Expected behavior**

To require  `@endcode` and fail

**Versions (please complete the following information):**
 - OS: MacOS 12.5.1
 - PHP: 8.1.7
 - PHPCS: 3.7.1
 - Standard: ?

**Additional context**

**Comment by claudiu-cristea** [comment]
Sorry, the rules are from other project
